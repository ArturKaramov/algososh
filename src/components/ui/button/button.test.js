import { Button } from "./button";
import React from "react";
import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Testing Button", () => {
  it("Button with text", () => {
    const tree = renderer.create(<Button text="Testing" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Disabled Button", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Callback on Button", () => {
    window.alert = jest.fn();
    render(<Button onClick={alert("Testing")} text="Testing" />);
    const button = screen.getByText("Testing");
    fireEvent.click(button);
    expect(window.alert).toHaveBeenCalledWith("Testing");
  });
});
