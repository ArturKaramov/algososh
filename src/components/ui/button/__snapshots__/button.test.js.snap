// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Testing
  </p>
</button>
`;

exports[`Testing Button Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing Button Disabled Button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
